Add Paris' post on communication (#20393)
kubernetes/website: content/en/blog/_posts/2020-04-21-contributor-communication-upstream-marketing.md

---
layout: blog
title: How Kubernetes contributors are building a better communication process
date: 2020-04-21
slug: contributor-communication
---

**Authors:** Paris Pittman

> “Perhaps we just need to use a different word. We may need to use community development or project advocacy as a word in the open source realm as opposed to marketing, and perhaps then people will realize that they need to do it.” 
> ~ [*Nithya Ruff*](https://todogroup.org/www.linkedin.com/in/nithyaruff/) (from [*TODO Group*](https://todogroup.org/guides/marketing-open-source-projects/))

A common way to participate in the Kubernetes contributor community is
to be everywhere. 

> We have an active [Slack], many mailing lists, Twitter account(s), and

Reviewer: ```suggestion
We have an active [Slack](https://slack.k8s.io), many mailing lists, Twitter account(s), and
```

dozens of community-driven podcasts and newsletters that highlight all

> end-user, contributor, and ecosystem topics. And to add on to that, we also have [repositories of amazing documentation], tons of [meetings] that drive the project forward, and [recorded code deep dives]. All of this information is incredibly valuable,

Reviewer: ```suggestion
end-user, contributor, and ecosystem topics. And to add on to that, we also have [repositories of amazing documentation](https://github.com/kubernetes/community), tons of [meetings](https://calendar.google.com/calendar/embed?src=cgnt364vd8s86hr2phapfjc6uk%40group.calendar.google.com&ctz=America%2FLos_Angeles) that drive the project forward, and [recorded code deep dives](https://www.youtube.com/watch?v=yqB_le-N6EE). All of this information is incredibly valuable,
```

- 52 mailing lists (kubernetes-dev@ has thousands of members; sig-networking@ has 1000 alone)
- 40 [community groups](https://github.com/kubernetes/community/blob/master/governance.md#community-groups)
- 30 [upstream meetings](https://calendar.google.com/calendar/embed?src=cgnt364vd8s86hr2phapfjc6uk%40group.calendar.google.com&ctz=America%2FLos_Angeles) *this* week alone

The scale is growing, and with that comes the need to ensure contributors are able to get the information lifed up front-and-center.

## How we got here

Kubernetes (k8s for short) communication grew out of a need for people
to connect in our growing community. With the best of intentions, the
community spun up channels for people to connect. This energy was part
of what helped Kubernetes grow so fast, and it also had us in sprawling out far and wide. As adoption grew, [contributors knew there was a need for standardization](https://github.com/kubernetes/community/issues/2466).

This new attention to how the community communicates led to a discovery
of a complex web of options. There were so many options, it was a
challenge for anyone to be sure they were in the right place to receive

> the right information. We started taking immediate action by pruning the

Reviewer: ```suggestion
the right information. We started taking immediate action combining communication streams and thinking about how to best reach out to serve our community.
```

> inactive mailing lists and other platforms, combining where we could,

Reviewer: ```suggestion
```

Author: ACK - this is in another commit.

> and thinking about the future. We also asked for feedback from all our

Reviewer: ```suggestion
We also asked for feedback from all our
```

  - Creating a presence on Twitter specifically for contributors via
    @k8scontributors that is all about being a contributor in all its
    forms.

What does this look like in the wild? Our [first post](https://kubernetes.io/blog/2020/03/19/join-sig-scalability/) in a series about our 36 community groups landed recently. Did you see it?
More articles like this and additional themes of stories to flow through
[our storytellers](https://github.com/kubernetes/community/tree/master/communication/marketing-team#purpose).

We will deliver this with an [ethos](https://github.com/kubernetes/community/blob/master/communication/marketing-team/CHARTER.md#ethosvision) behind us aligned to the Kubernetes project as whole, and we’re
committed to using the same tools as all the other SIGs to do so. Check out our [project board](https://github.com/orgs/kubernetes/projects/39) to view our roadmap of upcoming work.

## Join us and be part of the story

This initiative is early on and we still have important roles to fill to make it successful. 

If you are interested in open sourcing marketing functions – it’s a fun
ride – join us! Specific immediate roles include storytelling through

> blogs and designers and anything in progress on our project board. 

Reviewer: ```suggestion
blogs, designers, and anything in progress on our project board. 
```

to be everywhere. 

We have an active Slack, many mailing lists, Twitter account(s), and
dozens of community-driven podcasts and newsletters that highlight all
end-user, contributor, and ecosystem topics. And guess what else - an
entire repository of amazing documentation that I bet there is one gem
you didn’t know we had! All of this information is incredibly valuable,
but it can be too much. 

Keeping up with thousands of contributors can be a challenge for anyone.
but this task of consuming information straight from the firehose is
particularly challenging for new community members. It’s no secret that
the project is vast for contributors and users alike. 

To paint a picture with numbers: 

- 43,000 contributors
- 6,579 members in #kubernetes-dev slack channel
- 52 mailing lists (kubernetes-dev@ has thousands of members; sig-networking@ has 1000 alone)
- 40 [community groups](https://github.com/kubernetes/community/blob/master/governance.md#community-groups)
- 30 [upstream meetings](https://calendar.google.com/calendar/embed?src=cgnt364vd8s86hr2phapfjc6uk%40group.calendar.google.com&ctz=America%2FLos_Angeles) *this* week alone

> The scale is growing, and with that comes the need for attention.

Reviewer 2: How about something like "The scale of the project is growing, and with that comes the need to ensure contributors are able to get information about important tasks at hand.", or "important information is lifted up front-and-center for contributors" ?

Reviewer: thanks - went with your suggestion and combo'ed those

## How we got here

Kubernetes (k8s for short) communication grew out of a need for people
to connect in our growing community. With the best of intentions, the
community spun up channels for people to connect. This energy was part
of what helped Kubernetes grow so fast; it led us to sprawling out far and wide.

> sprawling out far and wide. As adoption grew, [contributors knew there was a need for standardization](https://github.com/kubernetes/community/issues/2466).

Reviewer 2: Repeat of "sprawling out far and wide."

Reviewer: thanks for the catch

This new attention to how the community communicates led to the discovery
of a complex web of options. With so many choices, it was a
challenge for anyone to be sure they were in the right place to receive
the right information. We started taking immediate action by pruning the
inactive mailing lists and other platforms, combining where we could,
and thinking about the future. We sought feedback from our

> contributors directly [**via annual surveys**](https://github.com/kubernetes/community/tree/master/sig-contributor-experience/surveys)

Reviewer 3: nit, don't let this block a merge
I'd write this as
```suggestion
contributors directly via [**annual surveys**](https://github.com/kubernetes/community/tree/master/sig-contributor-experience/surveys)
```

only imagine what it’s like getting important messages across to make
sure they are landing, and folks are taking action.

## Contributing to better communication

Think about how your company/employer solves for this. Many have done so
by building internal marketing and communication focus areas in
marketing departments. So that’s what we are doing. This has also been
applied [at Fedora](https://fedoraproject.org/wiki/Marketing) and at a smaller
scale in our very [own release](https://github.com/kubernetes/sig-release/tree/master/release-team) and [contributor summit](https://github.com/kubernetes/community/blob/d0fd6c16f7ee754b08082cc15658eb8db7afeaf8/events/events-team/marketing/README.md) planning
teams as roles. 

We have hit the accelerator on an **upstream marketing group** under SIG
[Contributor Experience](http://git.k8s.io/community/sig-contributor-experience), and we want to tackle this challenge straight
on. We’ve learned in other contributor areas that creating roles for
contributors is super helpful - onboarding, breaking down work, and
ownership. [Here’s our team charting the course](https://github.com/kubernetes/community/tree/master/communication/marketing-team).

> Journey your way through our other documents like our charter if you are

Reviewer 3: How about making “charter” a hyperlink?

interested in our mission and scope.

Many of you close to the ecosystem might be scratching your head - isn’t
this what CNCF does?

Yes and no. The CNCF has 40+ other projects that need to be marketed to
a countless number of different types of community members in distinct
ways, and they aren’t responsible for the day to day operations of their
projects. They absolutely do partner with us to highlight what we need
and when we need it, and they do a fantastic job of it (one example is
the [*@kubernetesio twitter account*](https://twitter.com/kubernetesio?lang=en) and its 200,000
followers). 

Where this working group differs is in its scope: we are entirely
focused on elevating the hard work being done throughout the Kubernetes
community by its contributors.

## What to expect from us

You can expect to see us in our web of [communication channels](https://github.com/kubernetes/community/tree/master/communication) supporting you by:

  - Finding ways of adding our human touch to potentially overwhelming
    quantities of info by storytelling and other methods - we want to
    highlight you and provide useful information!

>   - Providing ways to clarify expectations on what’s important, urgent,
>     or otherwise

Reviewer 3: nit:
```suggestion
  - Providing ways to clarify expectations on what’s important, urgent,
    or otherwise.
```

  - Keeping you in the know of the comings and goings of contributor
    community events, activities, mentoring initiatives, KEPs, and more.
  - Creating a presence on Twitter specifically for contributors via
    @k8scontributors that is all about being a contributor in all its
    forms.

> What does this look like in the wild? [Our first post](https://kubernetes.io/blog/2020/03/19/join-sig-scalability/) in a series about our 40 community groups landed recently. Did you see it?

Reviewer 3: nit, don't let this block a merge:
```suggestion
What does this look like in the wild? Our [first post](https://kubernetes.io/blog/2020/03/19/join-sig-scalability/) in a series about our 40 community groups landed recently. Did you see it?
```

> What does this look like in the wild? [Our first post](https://kubernetes.io/blog/2020/03/19/join-sig-scalability/) in a series about our 40 community groups landed recently. Did you see it?

Reviewer 2: Reword "Did you see it?" to "What did you think about it?"

More articles like this and additional themes of stories to flow through
[our storytellers](https://github.com/kubernetes/community/tree/master/communication/marketing-team#purpose).

We will deliver this with an [ethos](https://github.com/kubernetes/community/blob/master/communication/marketing-team/CHARTER.md#ethosvision) behind us aligned to the Kubernetes project as a whole, and we’re
committed to using the same tools as all the other SIGs to do so. Check out our [project board](https://github.com/orgs/kubernetes/projects/39) to view our roadmap of upcoming work.

## Join us and be part of the story

> This initiative is early on, and we still have important roles to fill to

Reviewer 2: "is in an early phase"

Reviewer: thanks!

make it successful. 

> If you are interested in open sourcing marketing functions – it’s a neat

Reviewer 2: I'd say "it's going to be a fun ride!" :)

---
layout: blog
title: How Kubernetes contributors are building a better communication process
date: 2020-04-21
slug: contributor-communication
---

**Authors:** Paris Pittman

> “Perhaps we just need to use a different word. We may need to use community development or project advocacy as a word in the open source realm as opposed to marketing, and perhaps then people will realize that they need to do it.” 
> ~ [*Nithya Ruff*](https://todogroup.org/www.linkedin.com/in/nithyaruff/) (from [*TODO Group*](https://todogroup.org/guides/marketing-open-source-projects/))

A common way to participate in the Kubernetes contributor community is
to be everywhere. 

We have an active Slack, many mailing lists, Twitter account(s), and
dozens of community-driven podcasts and newsletters that highlight all
end-user, contributor, and ecosystem topics. And guess what else - an

> entire repository of amazing documentation that I bet there is one gem

Reviewer: [an entire repository](https://github.com/kubernetes/community)

Reviewer: it would be cool to link the repo here as well

Reviewer: does "guess what else come off" as rude? that wasn't my intention at all - more cheery but I don't think it reads that way. worst case scenario, the sentence can lead without it.

Reviewer 4: nit: I agree that this could be separated into two sentences.
Guess what ... -- a SIG and repository dedicated to ... Kubernetes documentation.

Reviewer 2: Perhaps something like "And to add on top all of that, we also have repositories of amazing documentation, tons of meetings that drive the project forward, and recorded code deep dives." ?

Reviewer: went with @jonasrosland suggestion and included some links ex: [meetings] with calendar - PR coming soon

you didn’t know we had! All of this information is incredibly valuable,
but it can be too much. 

Keeping up with thousands of contributors can be a challenge for anyone,
but this task of consuming information straight from the firehose is
particularly challenging for new community members. It’s no secret that
the project is vast for contributors and users alike. 

To paint a picture with numbers: 

- 43,000 contributors
- 6,579 members in #kubernetes-dev slack channel
- 52 mailing lists (kubernetes-dev@ has thousands of members; sig-networking@ has 1000 alone)
- 40 [community groups](https://github.com/kubernetes/community/blob/master/governance.md#community-groups)
- 30 [upstream meetings](https://calendar.google.com/calendar/embed?src=cgnt364vd8s86hr2phapfjc6uk%40group.calendar.google.com&ctz=America%2FLos_Angeles) *this* week alone

The scale is growing, and with that comes the need for attention.

## How we got here

Kubernetes (k8s for short) communication grew out of a need for people
to connect in our growing community. With the best of intentions, the
community spun up channels for people to connect. This energy was part

> of what helped Kubernetes grow so fast, and it also had us in a space of

Reviewer 5: ```suggestion
of what helped Kubernetes grow so fast, and it also had us in the space of
```

sprawling out far and wide. As adoption grew, [contributors knew there was a need for standardization](https://github.com/kubernetes/community/issues/2466).

> This new attention to how the community communicates led to a discovery

Reviewer 5: ```suggestion
This new attention to how the community communicates led to the discovery
```

> of a complex web of options. There were so many options, it was a

Reviewer 5: ```suggestion
of a complex web of options. There were so many options, and it was a
```

challenge for anyone to be sure they were in the right place to receive
the right information. We started taking immediate action by pruning the

> inactive mailing lists and other platforms, combining where we could,

Reviewer 4: nit: combining what?

Reviewer: mailing lists and other forums (reducing redundancy, stale, etc)

Reviewer 4: Thinking of something like:
  ... combining communication streams [or channels] and thinking  
about how to [best reach out to [serve] our community?].

Author: That sentence was ambiguous - added "them" to flow.

and thinking about the future. We also asked for feedback from all our
contributors directly [**via annual surveys**](https://github.com/kubernetes/community/tree/master/sig-contributor-experience/surveys)
to see where folks were actually reading the news that influences their
experiences here in our community.

![Kubernetes channel access](https://user-images.githubusercontent.com/1744971/79478603-3a3a1980-7fd1-11ea-8b7a-d36aac7a097b.png)

> Our contributor base is larger than many enterprise companies. You can

Reviewer 5: ```suggestion
Our contributor base is more abundant than many enterprise companies. You can
```

Author: I like that edit @mrbobbytables -- noting the size is the intention, so a throwback to the specifics hits home.

Reviewer: ack'ing the change

only imagine what it’s like getting important messages across to make

> sure they are landing and folks are taking action. 

Reviewer 5: ```suggestion
sure they are landing, and folks are taking action.
```

## Contributing to better communication

> Think about how your company/employer solves for this. Many have done so

Reviewer 4: nit: Think about how your company/employer solves for this
What is solved?

Reviewer: Think about how your company or employer would tackle this? 
Solving: a communications strategy, promotions, creative, etc.

Reviewer 4: Think about how your company or employer solves communications [challenges, overflow?].
Companies build internal marketing and ... departments.

Author: Suggesting "Think about how your company/employer solves for this kind of communication challenge"

marketing departments. So that’s what we are doing. This has also been
applied [at Fedora](https://fedoraproject.org/wiki/Marketing) and at a smaller
scale in our very [own release](https://github.com/kubernetes/sig-release/tree/master/release-team) and [contributor summit](https://github.com/kubernetes/community/blob/d0fd6c16f7ee754b08082cc15658eb8db7afeaf8/events/events-team/marketing/README.md) planning
teams as roles. 

We have hit the accelerator on an **upstream marketing group** under SIG
Contributor Experience, and we want to tackle this challenge straight
on. We’ve learned in other contributor areas that creating roles for
contributors is super helpful - onboarding, breaking down work, and
ownership. [Here’s our team charting the course](https://github.com/kubernetes/community/tree/master/communication/marketing-team).

Journey your way through our other documents like our charter if you are
interested in our mission and scope.

Many of you close to the ecosystem might be scratching your head - isn’t
this what CNCF does?

Yes and no. The CNCF has 40+ other projects that need to be marketed to
a countless number of different types of community members in distinct

> ways and they aren’t responsible for the day to day operations of their

Reviewer 5: ```suggestion
ways, and they aren’t responsible for the day to day operations of their
```

projects. They absolutely do partner with us to highlight what we need
and when we need it, and they do a fantastic job of it (one example is
the [*@kubernetesio twitter account*](https://twitter.com/kubernetesio?lang=en) and its 200,000
followers). 

> Where this working group differs is in its scope: we are entirely

Reviewer: nix the working in "working group" here as it sounds too close to our [full fledged working groups](https://github.com/kubernetes/community/blob/master/sig-list.md#master-working-group-list) after reading. Ours is a part of a subproject of contribex (which sounds long and complex for this - you tell me)

Reviewer 6: Maybe just subproject? or group? "Where this group differs"

Reviewer 2: I would say subproject, under ContribEx

Reviewer: i went with group because the sentiment is more group than subproject of contribex etc

focused on elevating the hard work being done throughout the Kubernetes
community by its contributors.

## What to expect from us

> You can expect to see us in our web of [communication channels](https://github.com/kubernetes/community/tree/master/communication) supporting you by:

Reviewer 4: nit: `web of communication channels` or just
`in our communication channels and [where else ?]

Reviewer: fixed

  - Finding ways of adding our human touch to potentially overwhelming
    quantities of info by storytelling and other methods - we want to

>     highlight you and provide useful information!

Reviewer 4: nit: `we want to highlight you` or are you saying,
we want to focus on [you] the contributors (the humans)?

Reviewer: how about "highlight the work you are doing"

  - Providing ways to clarify expectations on what’s important, urgent,
    or otherwise
  - Keeping you in the know of the comings and goings of contributor
    community events, activities, mentoring initiatives, KEPs, and more.
  - Creating a presence on Twitter specifically for contributors via
    @k8scontributors that is all about being a contributor in all its
    forms.

What does this look like in the wild? [Our first post](https://kubernetes.io/blog/2020/03/19/join-sig-scalability/) in a series about our 40 community groups landed recently. Did you see it?
More articles like this and additional themes of stories to flow through
[our storytellers](https://github.com/kubernetes/community/tree/master/communication/marketing-team#purpose).

> We will deliver this with an [ethos](https://github.com/kubernetes/community/blob/master/communication/marketing-team/CHARTER.md#ethosvision) behind us aligned to the Kubernetes project as whole, and we’re

Reviewer 5: ```suggestion 
We will deliver this with an [ethos](https://github.com/kubernetes/community/blob/master/communication/marketing-team/CHARTER.md#ethosvision) behind us aligned to the Kubernetes project as a whole, and we’re
```

committed to using the same tools as all the other SIGs to do so. Check out our [project board](https://github.com/orgs/kubernetes/projects/39) to view our roadmap of upcoming work.

## Join us and be part of the story

> This initiative is early on and we still have important roles to fill to

Reviewer 5: ```suggestion
This initiative is early on, and we still have important roles to fill to
```

make it successful. 

If you are interested in open sourcing marketing functions – it’s a neat

> ride – join us! Specific immediate roles include storytelling through

Reviewer 6: ```suggestion
ride – [join us!](https://github.com/kubernetes/community/tree/master/communication/marketing-team#contact-us) Specific immediate roles include storytelling through
```

I know this is linked to later, but figure should make it easy if they want to join 👍

blogs and designers and anything in progress on our project board. 

> 

Reviewer 4: nit: Specific immediate roles include storytelling through blogs, designers, and anything in progress on our project board?

Author: Agreed - adjusted.

Also, if you’re reading this, you’re exactly the type of person we are
here to support. Want to give us feedback or suggestions? Missed

> something in the past that should have been highlighted? Contributor

Reviewer 5: ```suggestion
something in the past that should have been highlighted? A contributor
```

> that wants something you are working on to be highlighted? We would love

Reviewer 4: nit: Duplicate of previous line:
Contributor that wants something you are working on to be highlighted?

Reviewer: redid the whole paragraph to make it flow better:
"Also, if you’re reading this, you’re exactly the type of person we are
here to support. We would love to hear about how to improve, feedback, or how we can work together."

> to hear about how to improve. 

Reviewer: could we nix the last line? if we put the ending statement with the missed something in the past bit and end it there it will flow a bit better.
